python/pyarrow/compute.py

```diff
@@ -155,7 +157,7 @@ def _handle_options(name, option_class, options, kwargs):
             "Function {!r} expected a {} parameter, got {}"
             .format(name, option_class, type(options)))
 
-    return options
+    return option_class()
```

Comment:
(you basically get here when `options` is None, and there are also no `**kwargs`)

This seems a sensible change to me (use the default constructor of the options class when no options are specified), but I think it might also be an indication that on the C++ side the sort_indices kernel has not a default options set, which is something we might want to fix on the C++ side? (since all other classes seem to have this, and can be called without options class specified)

Now, this change also causes some failure in another test (one that tests the error message with wrong number of arguments, because it now errors here first). So if we keep it like this, I need to update that test

Reply:
I'm OK that we add `SortOptins::Defaults()` with empty `sort_keys` and use it for the default option of `sort_indices`.
It doesn't work for `RecordBatch` and `Table` but works for `Array` and `ChunkedArray`.

Reply:
Indeed, and for RecordBatch/Table, you already get a nice error message when providing it empty sort keys.